tests/es5.js

```diff
@@ -1,200 +1,115 @@
-
-var reconcileBlockHistory = require("../source/callback-style-wrappers").reconcileBlockHistoryWithCallback;
-var reconcileBlocksAndLogs = require("../source/callback-style-wrappers").reconcileBlocksAndLogsWithCallback;
+var BlockAndLogStreamer = require("../source/index").BlockAndLogStreamer;
+var getBLockByHashFactory = require("./helpers").getBlockByHashFactory;
+var getLogsFactory = require("./helpers").getLogsFactory;
 var MockBlock = require("./helpers").MockBlock;
 var MockLog = require("./helpers").MockLog;
 var getBlockByHashFactory = require("./helpers").getBlockByHashFactory;
 var getLogsFactory = require("./helpers").getLogsFactory;
 var expect = require("chai").expect;
-var ImmutableList = require("immutable").List;
 
-describe("reconcileBlockHistoryEs5", function () {
-  var newBlockAnnouncements;
-  var blockRemovalAnnouncments;
-  function onBlockAdded(block, callback) { setTimeout(function () { newBlockAnnouncements.push(block); callback(); }); }
-  function onBlockRemoved(block, callback) { setTimeout(function () { blockRemovalAnnouncments.push(block); callback(); }); }
+describe("BlockAndLogStreamer", function () {
+  var blockAndLogStreamer;
+  var blockAddedAnnouncements;
+  var blockRemovedAnnouncements;
+  var logAddedAnnouncements;
+  var logRemovedAnnouncements;
+  var onBlockAdded = (block) => blockAddedAnnouncements.push(block);
+  var onBlockRemoved = (block) => blockRemovedAnnouncements.push(block);
+  var onLogAdded = (log) => logAddedAnnouncements.push(log);
+  var onLogRemoved = (log) => logRemovedAnnouncements.push(log);
 
   beforeEach(() => {
-    newBlockAnnouncements = [];
-    blockRemovalAnnouncments = [];
+    var getBlockByHash = getBlockByHashFactory();
+    wrappedGetBlockByHash = function (hash, callback) {
+      getBlockByHash(hash)
+        .then(block => callback(undefined, block))
+        .catch(error => callback(error, undefined));
+    };
+    var getLogs = getLogsFactory(1);
+    wrappedGetLogs = function (filters, callback) {
+      getLogs(filters)
+        .then(logs => callback(undefined, logs))
+        .catch(error => callback(error, undefined));
+    };
+    blockAndLogStreamer = BlockAndLogStreamer.createCallbackStyle(wrappedGetBlockByHash, wrappedGetLogs, { blockRetention: 5 });
```

Comment:
Example of callback style construction.  (the wrappers are just because the testing infrastructure uses promises, but the callback style construction doesn't accept promises and I didn't want to recreate the test infrastructure)